internal/utils/comparators/daemonsets/comparator_test.go

```diff
@@ -58,6 +59,15 @@ var _ = Describe("daemonset#AreSame", func() {
 			ok, _ := daemonsets.AreSame(current, desired)
 			Expect(ok).To(BeTrue())
 		})
+
+		It("should fail when the secret hash is different", func() {
+			current.Spec.Template.Annotations = map[string]string{
+				constants.AnnotationSecretHash: "foo",
+			}
+			ok, _ := daemonsets.AreSame(current, desired)
+
+			Expect(ok).To(BeFalse())
```

Comment:
Tests :+1: 

Should we also check the second return type?

```suggestion
			ok, reason := daemonsets.AreSame(current, desired)

			Expect(ok).To(BeFalse())
			Expect(reason).To(Equal("secretHash"))
```